test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,502 @@
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if csvName == "" {
+					e2e.Logf("Subscription %q doesn't have installed CSV, retrying...", expectedSubscriptionName)
+					return false, nil
+				}
+				e2e.Logf("Subscription %q has installed CSV: %s", expectedSubscriptionName, csvName)
+				return true, nil
+			})
+			o.Expect(waitVersion).NotTo(o.HaveOccurred(), "Timed out waiting for the ClusterServiceVersion to install")
+
+			csvName, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			waitCSV := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				csvStatus, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "clusterserviceversion", csvName, "-o=jsonpath={.status.phase}").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if csvStatus != "Succeeded" {
+					e2e.Logf("Cluster Service Version %s is not successful, retrying...", csvName)
+					return false, nil
+				}
+				e2e.Logf("Cluster Service Version %s has succeeded!", csvName)
+				return true, nil
+			})
+			o.Expect(waitCSV).NotTo(o.HaveOccurred(), "Cluster Service Version %s never reached succeeded status", csvName)
+
+			// get OSSM Operator deployment
+			waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				deployOSSM, err := oc.AdminKubeClient().AppsV1().Deployments(expectedSubscriptionNamespace).Get(context, "servicemesh-operator3", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if deployOSSM.Status.ReadyReplicas < 1 {
+					e2e.Logf("OSSM operator deployment %q is not ready, retrying...", deploymentOSSM)
+					return false, nil
+				}
+				e2e.Logf("OSSM operator deployment %q is ready", deploymentOSSM)
+				return true, nil
+			})
+			o.Expect(waitErr).NotTo(o.HaveOccurred(), "OSSM Operator deployment %q did not successfully deploy its pod", deploymentOSSM)
+
+			g.By("Confirm that ISTIO CR is created and in healthy state")
+			resource := types.NamespacedName{Namespace: "openshift-ingress", Name: "openshift-gateway"}
+
+			waitCR := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				istioStatus, errIstio := oc.AsAdmin().Run("get").Args("-n", resource.Namespace, "istio", resource.Name, "-o=jsonpath={.status.state}").Output()
+				o.Expect(errIstio).NotTo(o.HaveOccurred())
+				if istioStatus != "Healthy" {
+					e2e.Logf("Istio CR %s is not healthy, retrying...", resource.Name)
+					return false, nil
+				}
+				e2e.Logf("Istio CR %s is in Healthy state!", resource.Name)
```

Comment:
```suggestion
				e2e.Logf("Istio CR %s is healthy!", resource.Name)
```
Just to be consistent with the analogous `CatalogSource` message.